skxray/diff_roi_choice.py

```diff
@@ -157,58 +141,61 @@ def roi_rings(img_dim, calibrated_center, num_rings,
 
     Returns
     -------
-    mesh : nedarry
+    labels_grid : array
         indices of the required rings
-        shape is ([detector_size[0]*detector_size[1]], )
+        shape is ([image_shape[0]*image_shape[1]], )
 
-    ring_inds : ndarray
-        indices of the required rings
+    """
+    grid_values = _grid_values(image_shape, calibrated_center)
+
+    edges = rings_edges(num_rings, first_r, delta_r)
 
-    ring_vals : ndarray
-        edge values of the required  rings
+    # indices of rings
+    labels_grid = np.digitize(np.ravel(grid_values), np.array(edges),
+                              right=False)
+    # discard the indices greater than number of rings
+    labels_grid[labels_grid > num_rings] = 0
 
-    num_pixels : ndarray
-        number of pixels in each ring
+    return labels_grid.reshape(image_shape)
 
-    pixel_list : ndarray
-        pixel indices for the required rings
 
+def rings_edges(num_rings, first_r, delta_r):
     """
-    grid_values = _grid_values(img_dim, calibrated_center)
+    This function will provide the edge values of the rings when
+    there is a no step value between each ring
 
-    # last ring edge value
-    last_r = first_r + num_rings*(delta_r)
+    Parameters
+    ----------
+    num_rings : int
+        number of rings
 
-    # edges of all the rings
-    q_r = np.linspace(first_r, last_r, num=(num_rings+1))
+    first_r : float
+        radius of the first  ring
 
-    # indices of rings
-    mesh = np.digitize(np.ravel(grid_values), np.array(q_r),
-                            right=False)
-    # discard the indices greater than number of rings
-    mesh[mesh > num_rings] = 0
+    delta_r : float
+        thickness of the ring
 
-    # Edge values of each rings
-    ring_vals = []
+    Returns
+    -------
+    ring_vals : array
+        edge values of each ring
 
-    for i in range(0, num_rings):
-        if i < num_rings:
-            ring_vals.append(q_r[i])
-            ring_vals.append(q_r[i + 1])
-        else:
-            ring_vals.append(q_r[num_rings-1])
+    ring_edges : array
+        edge values of each ring
+        shape is (num_rings, 2)
+    """
 
-    ring_vals = np.asarray(ring_vals)
+    # last ring edge value
+    last_r = first_r + num_rings*(delta_r)
 
-    (ring_inds, ring_vals, num_pixels,
-     pixel_list) = _process_rings(num_rings, img_dim,
-                                  ring_vals, mesh)
+    # edges of all the rings
+    ring_edges = np.linspace(first_r, last_r, num=(num_rings+1))
 
-    return mesh.reshape(img_dim), ring_inds, ring_vals, num_pixels, pixel_list
+    return ring_edges
 
 
-def roi_rings_step(img_dim, calibrated_center, num_rings,
-               first_r, delta_r, *step_r):
+def rings_step(image_shape, calibrated_center, num_rings, first_r, delta_r,
+               *step_r):
```

Comment:
why is `step_r` a variadic?  It should probably be an list/array so that the call signature looks like 

```python
rings = rings_step(shape, center, first_r, delta_r, [1, 1, 1, 2, 1, 1])
```

rather than

```python
rings = rings_step(shape, center, first_r, delta_r, 1, 1, 1, 2, 1, 1)
```